src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,158 @@ def test_set_beam_shift(self):
+            self.assertEqual(autostigmator_state, True)
+            self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+
+            starting_time = time.time()
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_autostigmating(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > 30:
+                    break
+
+            # Check if the time to perform the autofocus is not to long
+            self.assertLess(time.time() - starting_time, 30, "Execution auto-stigmation is slow.")
+            # Line to inspect the execution time to update the expected time
+            print("Execution time was %s seconds " % (time.time() - starting_time))
+
+            stigmator_values_x.append(self.microscope.get_stigmator()[0])
+            stigmator_values_y.append(self.microscope.get_stigmator()[1])
+
+        # Check if all the stigmator values are within the required range each run
+        stigmator_range = self.microscope.stigmator_info()['range']
+        min_x_stig_values, max_x_stig_values = stigmator_range['x']
+        min_y_stig_values, max_y_stig_values = stigmator_range['y']
+        self.assertGreaterEqual(min(stigmator_values_x), min_x_stig_values)
+        self.assertLessEqual(max(stigmator_values_x), max_x_stig_values)
+        self.assertGreaterEqual(min(stigmator_values_y), min_y_stig_values)
+        self.assertLessEqual(max(stigmator_values_y), max_y_stig_values)
+
+        # TODO K.K. remove debug values after testing
+        print("X list is")
+        print(stigmator_values_x)
+        print("Y list is")
+        print(stigmator_values_y)
+        print("With a range of:")
+        print(self.microscope.stigmator_info()['range'])
+
+        # Test if an error is raised if an invalid detector is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoStigmator("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, False)  # Check if state remained unchanged
+
+    def test_apply_auto_contrast_brightness(self):
+        """
+        Test for the auto functionality of the apply_auto_contrast_brightness.
+        Only checks the returned state, the return object and if an error is raised for an incorrect input state.
+        """
+        for detector in detector2ChannelName:
+            auto_contrast_brightness_return = self.scanner.applyAutoContrastBrightness(detector)
+            time.sleep(2.5)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
```

Comment:
Add comments the same way as for the auto stigmation test.

Reply:
The stigmation test if slightly different (it first tests basic functionality and then after it repeats the test and compaires the outcome)
If I look at the other tests in this test file I don't see this extensive commenting. And to me the line "auto_contrast_brightness_future = self.scanner.applyAutoContrastBrightness(detector)" suggests that the "AutoContrastBrightness" function is applied and some lines down that:  "auto_contrast_brightness_future.cancel()" this cancels it.
But maybe that is just me not liking extra comments what do you guys think @Sabrina85  and @pieleric ?

Reply:
I asked this for consistency. If you do it in one test, do it also in the other ones. The reader easily can follow the structure then. The reader will also immediately notice the blocks that are different in the tests. Alternatively, don't do it in any test and only comment the "special part" in the autostigmator.

Reply:
I see and indeed I agree, lets discuss briefly during the STUP.